legend-engine-shared-core/src/main/java/org/finos/legend/engine/shared/core/api/grammar/GrammarAPI.java

```diff
@@ -103,6 +121,19 @@ protected <T> Response jsonToGrammar(T graphFetchTree, RenderStyle renderStyle,
         }
     }
 
+    protected <T> Response jsonToGrammarWithElements(T graphFetchTree, RenderStyle renderStyle, Function2<T, RenderStyle, Map<String, String>> func, ProfileManager<CommonProfile> pm, String spanText)
```

Comment:
let's rename `T graphFetchTree` to `input`, let's do it for the function above as well

Reply:
also, same comment as above about the need for this